packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -775,7 +775,7 @@ def _sender_act(self) -> Generator[None, None, None]:
                 self.context.logger.error(
                     "Trying to finalize a transaction which has been verified already!"
                 )
-            else:
+            else:  # pragma: no cover
```

Comment:
to painful to mock

Reply:
So we keep it untested?

Reply:
It can, of course, be done, though a small refactor would work wonders here. One may imagine all of the code after this line 
https://github.com/valory-xyz/open-autonomy/blob/bdcd9236d04b3198c0d414b634fd3a7dbf414ea2/packages/valory/skills/transaction_settlement_abci/behaviours.py#L767
and inside this context to be a function that would take the `tx_data` as an argument and return the `payload`. Not only would it be more readable, it would be much easier to test.